### PR TITLE
Bump stage0 to go1.22.6

### PR DESCRIPTION
**Pull request by qmuntal**
Upstream Go master branch requires at least go1.22.6 for bootstrapping: https://go-review.googlesource.com/c/go/+/606156.

Will help unblock #1285.

### CODE REVIEW

**Comment by dagood** [comment]
CodeQL error looks like it's the same as this one--different version in the issue title, but because all we updated in this PR is the Go version, I don't doubt it's the same root cause:

* https://github.com/github/codeql-action/issues/2467.

(I commented in the issue with a link to our run so they can confirm and hopefully help me understand when this will be fixed.)

**Comment by karianna** [comment]
So is our GH Action for CodeQL failure expected?

**Comment by dagood** [comment]
They don't see a connection to that issue, actually. They suggested re-running it in debug mode and filing a new one. I figured out how to do that and it's going now. (There's a checkbox when you trigger the run in a specific way. Unfortunately, I don't know which one, because I hit start before thinking to note it down. 😄)

Edit: this one:

> image

**Comment by dagood** [comment]
* Filed this issue: https://github.com/github/codeql/issues/17526

**Review by karianna — APPROVED** [review]

**Comment by karianna** [comment]
Feel free to merge if you're comfortable (I'm good with it, thanks for the explanation).

**Comment by qmuntal** [comment]
> Feel free to merge if you're comfortable (I'm good with it, thanks for the explanation).

I'll try to understand a little bit better the codeql issue before merging this PR just in case we are missing something important.

**Comment by qmuntal** [comment]
Submitted https://github.com/microsoft/go/pull/1329 with what I think a fix that will unblock this PR.

**Pull request merged**
